src/coreclr/jit/lclvars.cpp

```diff
@@ -772,12 +776,20 @@ void Compiler::lvaInitUserArgs(InitVarDscInfo* varDscInfo, unsigned skipArgs, un
             // Anything that follows will also be on the stack. However, if something from
             // floating point regs has been spilled to the stack, we can still use r0-r3 until they are full.
 
-            if (varDscInfo->canEnreg(TYP_INT, 1) &&       // The beginning of the struct can go in a register
-                !varDscInfo->canEnreg(TYP_INT, cSlots) && // The end of the struct can't fit in a register
-                varDscInfo->existAnyFloatStackArgs())     // There's at least one stack-based FP arg already
+            if (varDscInfo->canEnreg(TYP_INT, 1) &&     // The beginning of the struct can go in a register
+                !varDscInfo->canEnreg(TYP_INT, cSlots)) // The end of the struct can't fit in a register
             {
-                varDscInfo->setAllRegArgUsed(TYP_INT); // Prevent all future use of integer registers
-                preSpill = false;                      // This struct won't be prespilled, since it will go on the stack
+                // This is a candidate for being split assuming we pre spill
+                // this and that we did not already have float args.
+                if (varDscInfo->existAnyFloatStackArgs())
+                {
+                    varDscInfo->setAllRegArgUsed(TYP_INT); // Prevent all future use of integer registers
+                    preSpill = false; // This struct won't be prespilled, since it will go on the stack
+                }
+                else if (preSpill)
+                {
+                    varDscInfo->hasSplitParam = true;
+                }
```

Comment:
After #66367 there is a better place to put this -- inside the inner block here:
https://github.com/dotnet/runtime/blob/a4a6d30517b5b141b3934814100dfb5f20f07d5d/src/coreclr/jit/lclvars.cpp#L960-L973
Can you revert this and just put `varDscInfo->hasSplitParam = true;` in there?

Reply:
Fixed. Thank you!